utility_code/weekly_update_player_data.py

```diff
@@ -46,6 +46,11 @@ def process_plugin_data(plugin_data):
         if item_replace_manager.replace_item(item, log_dict=replacements_log, debug_path=item.get_path_str()):
             num_replacements += 1
 
+    # TODO: Eventually should refactor plugin data so it has a top-level iterator interface to eliminate the need to call charms() directly
+    for item in plugin_data.charms().recursive_iter_items():
```

Comment:
It looks like the charms are not serialized back into json when done updating, but otherwise all the required pieces are in place.